### PR TITLE
Editable Grid Improvements: Validation of fields - selenium tests

### PR DESCRIPTION
**Pull request by XingY**
#### Related Pull Requests
- https://github.com/LabKey/labkey-ui-components/pull/1527
- https://github.com/LabKey/labkey-ui-premium/pull/462
- https://github.com/LabKey/platform/pull/5653
- https://github.com/LabKey/limsModules/pull/429
- https://github.com/LabKey/testAutomation/pull/1974

#### Changes
*

### CODE REVIEW

**Review by labkey-chrisj — APPROVED** [review]
Looks good

File: src/org/labkey/test/tests/component/EditableGridTest.java [hunk 1]
```diff
@@ -505,6 +602,229 @@ private EditableGrid goToEditableGrid(String sampleType)
...
+        {
+            for (int col = 0; col < ALL_FIELD_NAMES.size(); col++)
+            {
+                String colName = ALL_FIELD_NAMES.get(col);
+                if (colName.endsWith(" Req"))
+                    colName += " *";
+
+                checker().verifyFalse("Cell warning should be absent when a row is added on page load", testGrid.hasCellWarning(row, colName));
+            }
+        }
```
Comment (labkey-danield) [inline src/org/labkey/test/tests/component/EditableGridTest.java]: Could you do this?

```
checker().verifyEquals("Cell warning should be absent when a row is added on page load", 
0, Locator.tagWithClass("div", "cell-warning").findElements(testGrid).size());
```

File: src/org/labkey/test/tests/component/EditableGridTest.java [hunk 2]
```diff
@@ -505,6 +602,229 @@ private EditableGrid goToEditableGrid(String sampleType)
...
+        mouseOver(testGrid.getCell(0, "Row"));
+        testGrid.setCellValue(1, STR_FIELD_NAME, "This value is too long");
+        mouseOver(testGrid.getCell(0, "Row"));
+        testGrid.setCellValue(0, REQ_STR_FIELD_NAME + " *", "good");
+        mouseOver(testGrid.getCell(0, "Row"));
+        testGrid.setCellValue(1, REQ_STR_FIELD_NAME + " *", "This value is too long");
+        checker().verifyEquals("Cell warning status not as expected at row " + 0 + " for col " + STR_FIELD_NAME, false, testGrid.hasCellWarning(0, STR_FIELD_NAME));
+        checker().verifyEquals("Cell warning msg not as expected at row " + 1 + " for col " + STR_FIELD_NAME, "22/10 characters", testGrid.getCellPopoverText(1, STR_FIELD_NAME));
+        checker().verifyEquals("Cell warning status not as expected at row " + 0 + " for col " + REQ_STR_FIELD_NAME, false, testGrid.hasCellWarning(0, REQ_STR_FIELD_NAME + " *"));
+        checker().verifyEquals("Cell warning msg not as expected at row " + 1 + " for col " + REQ_STR_FIELD_NAME, "22/10 characters", testGrid.getCellPopoverText(1, REQ_STR_FIELD_NAME + " *"));
```
Comment (labkey-danield) [inline src/org/labkey/test/tests/component/EditableGridTest.java:635]: Any value in getting a screenshot after these checks (if there is a failure)?

File: src/org/labkey/test/tests/component/EditableGridTest.java [hunk 3]
```diff
@@ -505,6 +602,229 @@ private EditableGrid goToEditableGrid(String sampleType)
...
+        {
+            for (int col = 0; col < ALL_FIELD_NAMES.size(); col++)
+            {
+                String colName = ALL_FIELD_NAMES.get(col);
+                if (colName.endsWith(" Req"))
+                    colName += " *";
+
+                checker().verifyFalse("Cell warning should be absent after clearing out all cell values", testGrid.hasCellWarning(row, colName));
+            }
+        }
```
Comment (labkey-danield) [inline src/org/labkey/test/tests/component/EditableGridTest.java]: Same question/comment as above.

**Review by labkey-danield — APPROVED** [review]

File: src/org/labkey/test/tests/component/EditableGridTest.java (hunk 1 above)
Comment (XingY) [reply src/org/labkey/test/tests/component/EditableGridTest.java]: good idea, this is much more efficient

File: src/org/labkey/test/tests/component/EditableGridTest.java (hunk 2 above)
Comment (XingY) [reply src/org/labkey/test/tests/component/EditableGridTest.java:635]: screenshots added

**Pull request merged**